app/uk/gov/hmrc/exports/mongock/changesets/CacheChangeLog.scala

```diff
@@ -16,144 +16,35 @@
-  @ChangeSet(order = "001", id = "Exports DB Baseline", author = "Paulo Monteiro")
-  def dbBaseline(db: MongoDatabase): Unit = {}
-
-  @ChangeSet(order = "002", id = "CEDS-2231 Change country name to country code for location page", author = "Patryk Rudnicki")
-  def updateAllCountriesNameToCodesForLocationPage(db: MongoDatabase): Unit = {
-    val queryParams: ImmutableMap[String, AnyRef] =
-      ImmutableMap.of("locations.goodsLocation.country", ImmutableMap.of("$exists", true, "$ne", ""))
-    val query: Document = new Document(queryParams)
-
-    val documents = db.getCollection(collection).find(new BasicDBObject(query)).toSeq
-
-    documents.foreach { document =>
-      val goodsLocation = document
-        .get("locations", classOf[Map[String, Map[String, String]]])
-        .get("goodsLocation")
-        .asInstanceOf[Map[String, String]]
-
-      val countryName: String = goodsLocation.get("country")
-
-      goodsLocation.put("country", findCountryCode(countryName))
-
-      // update document on MongoDB
-      val queryIndexes: Map[String, String] =
-        ImmutableMap.of("id", document.get("id").asInstanceOf[String], "eori", document.get("eori").asInstanceOf[String])
-
-      val objectToBeUpdated: BasicDBObject = new BasicDBObject(queryIndexes)
-
-      db.getCollection(collection).replaceOne(objectToBeUpdated, document)
-    }
-  }
-
-  private def findCountryCode(countryName: String): String = {
-    val service: CountriesService = new CountriesService
-
-    service.allCountries.find(_.countryName == countryName).map(_.countryCode).getOrElse(countryName)
-  }
-
-  @ChangeSet(order = "003", id = "CEDS-2247 Change origination country structure", author = "Patryk Rudnicki")
-  def changeOriginationCountryStructure(db: MongoDatabase): Unit = {
-    val queryParams: ImmutableMap[String, AnyRef] =
-      ImmutableMap.of("locations.originationCountry", ImmutableMap.of("$exists", true, "$ne", ""))
-    val query: Document = new Document(queryParams)
-
-    val documents = db.getCollection(collection).find(new BasicDBObject(query)).toSeq
+  @ChangeSet(order = "006", id = "CEDS-2250 Add one structure level to /transport/borderModeOfTransportCode", author = "Maciej Rewera")
+  def updateTransportBorderModeOfTransportCode(db: MongoDatabase): Unit = {
 
-    documents.foreach { document =>
-      val locations = document.get("locations", classOf[Map[String, String]])
-      val originationCountry: String = locations.get("originationCountry")
+    logger.info("Applying 'CEDS-2250 Add one structure level to /transport/borderModeOfTransportCode'...")
 
-      val queryIndexes: Map[String, String] =
-        ImmutableMap.of("id", document.get("id").asInstanceOf[String], "eori", document.get("eori").asInstanceOf[String])
+    getDeclarationsCollection(db).updateMany(
+      and(exists("transport.borderModeOfTransportCode"), not(Filters.eq("transport.borderModeOfTransportCode", ""))),
+      rename("transport.borderModeOfTransportCode", "temp")
```

Comment:
I'm wondering why you're changing this to temp and after that once again name? Is not easier to just retrieve the value and update it?

With this approach you're retrieving the collection two times.

Reply:
This is what you'd get:
`com.mongodb.MongoWriteException: The source and target field for $rename must not be on the same path: transport.borderModeOfTransportCode: "transport.borderModeOfTransportCode.code" `
So you have to create a temporary field.
And here, you are not retrieving collection at all. You only ask mongo to do update. It is true that it is twice, but still it is over 100 times faster than the solution with reading documents, updating them in the application and sending them for update to Mongo.